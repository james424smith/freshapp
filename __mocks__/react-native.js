import * as ReactNative from "react-native";

export * from "react-native";
export const Vibration = { vibrate: jest.fn() };
export const Adnimated = { spring: jest.fn() };

export default Object.setPrototypeOf(
  {
    Vibration,
    Adnimated,
  },
  ReactNative
);
