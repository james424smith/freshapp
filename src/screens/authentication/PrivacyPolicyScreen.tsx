import React, { useEffect } from "react";
import { ScrollView, View } from "react-native";
import styles from "../../styles/AuthStyles";
import PdfRender from "../../components/PdfRender";
import { PrivacyPolicy } from "../../interfaces";
import { useSelector, useDispatch } from "react-redux";
import { getPrivacyPolicyDetails } from "../../redux/actions";
import { IRootReducerType } from "../../redux/reducers";

const PrivacyPolicyScreen = () => {
  const dispatch = useDispatch();
  const privacyPolicy = useSelector<IRootReducerType, PrivacyPolicy>(
    ({ privacyPolicyDetails }) => privacyPolicyDetails.privacyPolicyDetails
  );

  useEffect(() => {
    dispatch(getPrivacyPolicyDetails());
  }, [dispatch]);

  const pdf = privacyPolicy?.document;

  return (
    <ScrollView contentContainerStyle={styles.scrollViewContainer}>
      {pdf ? <PdfRender document={pdf} resourceType={"base64"} /> : <View />}
    </ScrollView>
  );
};

export default PrivacyPolicyScreen;
