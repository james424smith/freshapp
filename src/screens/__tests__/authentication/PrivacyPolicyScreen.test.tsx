import React from "react";
import PrivacyPolicyScreen from "../../authentication/PrivacyPolicyScreen";
import { render, cleanup } from "@testing-library/react-native";
import { Provider } from "react-redux";
import configureMockStore from "redux-mock-store";

describe("PrivacyPolicyScreen snapshot test", () => {
  afterEach(() => {
    cleanup();
    jest.clearAllMocks();
  });

  it("should render correct ui when focus is true when privacyPolicyDocument is present", () => {
    const mockedState = {
      privacyPolicyDetails: {
        privacyPolicyDetails: {
          document: "1234",
        },
      },
      network: { isConnected: true },
    };
    const mockedStore = configureMockStore();

    const { toJSON } = render(
      <Provider store={mockedStore(mockedState)}>
        <PrivacyPolicyScreen />
      </Provider>
    );
    expect(toJSON()).toMatchSnapshot();
  });

  it("should render correct ui when focus is true when employmentOfferDocument is undefined", () => {
    const mockedState = {
      privacyPolicyDetails: {
        privacyPolicyDetails: {
          document: undefined,
        },
      },
      network: { isConnected: true },
    };
    const mockedStore = configureMockStore();

    const { toJSON } = render(
      <Provider store={mockedStore(mockedState)}>
        <PrivacyPolicyScreen />
      </Provider>
    );
    expect(toJSON()).toMatchSnapshot();
  });
});
