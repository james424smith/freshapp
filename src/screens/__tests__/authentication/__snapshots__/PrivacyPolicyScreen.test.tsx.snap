// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrivacyPolicyScreen snapshot test should render correct ui when focus is true when employmentOfferDocument is undefined 1`] = `
<RCTScrollView
  contentContainerStyle={
    Object {
      "flexGrow": 1,
    }
  }
>
  <View>
    <View />
  </View>
</RCTScrollView>
`;

exports[`PrivacyPolicyScreen snapshot test should render correct ui when focus is true when privacyPolicyDocument is present 1`] = `
<RCTScrollView
  contentContainerStyle={
    Object {
      "flexGrow": 1,
    }
  }
>
  <View>
    <View
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <PDFView
        fadeInDuration={10}
        onError={[Function]}
        onLoad={[Function]}
        onPageChanged={[Function]}
        onScrolled={[Function]}
        resource="1234"
        resourceType="base64"
        style={
          Object {
            "flex": 1,
            "width": 750,
          }
        }
        textEncoding="utf-8"
        urlProps={Object {}}
      />
    </View>
  </View>
</RCTScrollView>
`;
